src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3815 @@
+        auto const USD = gw["USD"];
+
+        env.fund(XRP(10000), alice, bob, gw);
+        env.close();
+
+        env.trust(USD(1000), alice, bob);
+        env(pay(gw, alice, USD(100)));
+        env(pay(gw, bob, USD(100)));
+        env.close();
+
+        // success
+        {
+            auto const aliceSeq = env.seq(alice);
+            auto const bobSeq = env.seq(bob);
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+            auto const preAliceUSD = env.balance(alice, USD.issue());
+            auto const preBobUSD = env.balance(bob, USD.issue());
+
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            uint256 const chkID{getCheckIndex(bob, env.seq(bob))};
+            auto const [txIDs, batchID] = submitBatch(
+                env,
+                tesSUCCESS,
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(check::create(bob, alice, USD(10)), bobSeq),
+                batch::inner(check::cash(alice, chkID, USD(10)), aliceSeq + 1),
+                batch::sig(bob));
+            env.close();
+
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", batchID, std::nullopt},
+                {1, "CheckCreate", "tesSUCCESS", txIDs[0], batchID},
+                {2, "CheckCash", "tesSUCCESS", txIDs[1], batchID},
+            };
+            validateClosedLedger(env, testCases);
+
+            // Alice consumes sequences (# of txns)
+            BEAST_EXPECT(env.seq(alice) == aliceSeq + 2);
+
+            // Alice consumes sequences (# of txns)
+            BEAST_EXPECT(env.seq(bob) == bobSeq + 1);
+
+            // Alice pays Fee; Bob XRP Unchanged
+            BEAST_EXPECT(env.balance(alice) == preAlice - batchFee);
+            BEAST_EXPECT(env.balance(bob) == preBob);
+
+            // Alice pays USD & Bob receives USD
+            BEAST_EXPECT(
+                env.balance(alice, USD.issue()) == preAliceUSD + USD(10));
+            BEAST_EXPECT(env.balance(bob, USD.issue()) == preBobUSD - USD(10));
+        }
+
+        // failure
+        {
+            env(fset(alice, asfRequireDest));
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const bobSeq = env.seq(bob);
+            auto const preAlice = env.balance(alice);
+            auto const preBob = env.balance(bob);
+            auto const preAliceUSD = env.balance(alice, USD.issue());
+            auto const preBobUSD = env.balance(bob, USD.issue());
+
+            auto const batchFee = batch::calcBatchFee(env, 1, 2);
+            uint256 const chkID{getCheckIndex(bob, env.seq(bob))};
+            auto const [txIDs, batchID] = submitBatch(
+                env,
+                tesSUCCESS,
+                batch::outer(alice, aliceSeq, batchFee, tfIndependent),
+                batch::inner(check::create(bob, alice, USD(10)), bobSeq),
```

Comment:
```suggestion
                // tecDST_TAG_NEEDED - alice has enabled asfRequireDest
                batch::inner(check::create(bob, alice, USD(10)), bobSeq),
```
I found the comments in all the other tests super helpful, so just doing the same here